src/rascal/structure_managers/adaptor_increase_maxorder.hh

```diff
@@ -491,55 +518,6 @@ namespace rascal {
     max_cluster_indices.fill_sequence();
   }
 
-  /* ---------------------------------------------------------------------- */
-  /**
-   * Returns the linear indices of the clusters (whose atom tags are stored
-   * in counters). For example when counters is just the list of atoms, it
-   * returns the index of each atom. If counters is a list of pairs of indices
-   * (i.e. specifying pairs), for each pair of indices i,j it returns the
-   * number entries in the list of pairs before i,j appears.
-   */
-  template <class ManagerImplementation>
-  template <size_t Order>
-  size_t AdaptorMaxOrder<ManagerImplementation>::get_offset_impl(
-      const std::array<size_t, Order> & counters) const {
-    static_assert(Order < traits::MaxOrder,
-                  "this implementation handles only up to the respective"
-                  " MaxOrder");
-    // Order accessor: 0 - atoms
-    //                 1 - pairs
-    //                 2 - triplets
-    //                 etc.
-    // Order is determined by the ClusterRef building iterator, not by the
-    // Order of the built iterator
-
-    // necessary construct for static branching
-    using IncreaseHelper_t =
-        internal::IncreaseHelper<Order == (traits::MaxOrder - 1)>;
-
-    if (Order < (traits::MaxOrder - 1)) {
-      // If not accessible at this order, call lower Order offsets from lower
-      // order manager or push through to lower levels, if adaptors are
-      // stacked.
-      return IncreaseHelper_t::get_offset(*this->manager, counters);
-    } else {
-      // Counters is an array to call parent offset multiplet. This can then
-      // be used to access the actual offset for the Order which was built
-      // here. It needs to be cast into a smaller one to access the order of
-      // this Cluster(Order-1) from the manager below
-      std::array<size_t, Order - 1> counters_below{};
-      for (size_t c_index{0}; c_index < Order - 1; ++c_index) {
-        counters_below[c_index] = counters[c_index];
-      }
-      // Linear index of the Cluster (Order-1)
-      auto i{this->manager->get_offset_impl(counters_below)};
-      // Number of cluster in its current iteration
-      auto j{counters[Order - 1]};
```

Comment:
The new `get_offset_impl` seems much cleaner. Is it because we do not rely on any info from the previous offsets anymore? It seems _too easy_

Reply:
yes either it has all the data to build the cluster of order == MaxOrder or it asks for the previous manager to get the data to build a cluster of order < MaxOrder. I like it simple.

Reply:
Ok, to paraphrase: since we only _directly_ ask for a specific cluster of Order > 1, there is no need for the intermediate pair iteration, when we go to triplets, right?

Reply:
yes